ferry.go

```diff
@@ -1158,3 +1162,49 @@ func (f *Ferry) checkSourceForeignKeyConstraints() error {
 
 	return nil
 }
+
+func (f *Ferry) GetTotalRowsAndBytesMap() (totalRowsPerTable map[string]uint64, totalBytesPerTable map[string]uint64) {
+	totalRowsPerTable = make(map[string]uint64)
+	totalBytesPerTable = make(map[string]uint64)
+
+	for _, table := range f.Tables {
+		query := fmt.Sprintf(`
+		SELECT table_rows, data_length
+		FROM information_schema.tables
+		WHERE table_schema='%s'
+		AND table_name='%s'
+		`,
+			table.Schema,
+			table.Name,
+		)
+
+		rows, err := f.SourceDB.Query(query)
+		if err != nil {
+			return
+		}
+		defer rows.Close()
+
+		var totalRows sqlorig.NullInt64
+		var totalBytes sqlorig.NullInt64
+
+		for rows.Next() {
+			if err = rows.Scan(&totalRows, &totalBytes); err != nil {
```

Comment:
Is there a reason why you're not returning the error from here and making the stat values as 0 ?

Reply:
Errors happened while scanning would have nothing to do with issues in connection. Although I get your reasoning on why you're not returning from the error, but just for the sake of consistency with codebase we should return the errors.

Reply:
Update: Paired with Manan